lib/api/model/initial_snapshot.dart

```diff
@@ -102,3 +112,54 @@ class RecentDmConversation {
 
   Map<String, dynamic> toJson() => _$RecentDmConversationToJson(this);
 }
+
+/// The `user_settings` dictionary.
+///
+/// For docs, search for "user_settings:"
+/// in <https://zulip.com/api/register-queue>.
+@JsonSerializable(fieldRename: FieldRename.snake, createFieldMap: true)
+class UserSettings {
+  bool twentyFourHourTime;
+  bool? displayEmojiReactionUsers; // TODO(server-6)
+
+  // TODO more, as needed. When adding a setting here, please also:
+  // (1) add it to the [UserSettingName] enum below
+  // (2) then re-run the command to refresh the .g.dart files
+  // (3) handle the event that signals an update to the setting
+
+  UserSettings({
+    required this.twentyFourHourTime,
+    this.displayEmojiReactionUsers,
```

Comment:
nit:
```suggestion
    required this.twentyFourHourTime,
    required this.displayEmojiReactionUsers,
```

I think we do want any callsite to be explicit about making this `null`, if it so chooses.

(Any callsite is going to be in a test, other than the generated one which always passes all arguments anyway.)